flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/SortUtil.scala

```diff
@@ -0,0 +1,247 @@
+import org.apache.flink.api.common.operators.Order
+import org.apache.calcite.rex.{RexLiteral, RexNode}
+import org.apache.flink.api.common.ExecutionConfig
+import org.apache.flink.api.common.typeinfo.AtomicType
+import org.apache.flink.api.java.typeutils.runtime.RowComparator
+import org.apache.flink.api.common.typeutils.TypeSerializer
+import org.apache.flink.table.runtime.types.{CRow, CRowTypeInfo}
+
+/**
+ * Class represents a collection of helper methods to build the sort logic.
+ * It encapsulates as well the implementation for ordering and generic interfaces
+ */
+object SortUtil {
+
+  /**
+   * Function creates [org.apache.flink.streaming.api.functions.ProcessFunction] for sorting 
+   * elements based on rowtime and potentially other fields
+   * @param collationSort The Sort collation list
+   * @param inputType input row type
+   * @param execCfg table environment execution configuration
+   * @return org.apache.flink.streaming.api.functions.ProcessFunction
+   */
+  private[flink] def createRowTimeSortFunction(
+    collationSort: RelCollation,
+    inputType: RelDataType,
+    inputTypeInfo: TypeInformation[Row],
+    execCfg: ExecutionConfig): ProcessFunction[CRow, CRow] = {
+
+    val keySortFields = getSortFieldIndexList(collationSort)
+    //drop time from comparison as we sort on time in the states and result emission
+    val keyIndexesNoTime = keySortFields.slice(1, keySortFields.size)
+    val booleanOrderings = getSortFieldDirectionBooleanList(collationSort)
+    val booleanDirectionsNoTime = booleanOrderings.slice(1, booleanOrderings.size)
+    
+    val fieldComps = createFieldComparators(
+      inputType, 
+      keyIndexesNoTime, 
+      booleanDirectionsNoTime,
+      execCfg)
+    val fieldCompsRefs = fieldComps.asInstanceOf[Array[TypeComparator[AnyRef]]]
+    
+    val rowComp = new RowComparator(
+       inputType.getFieldCount,
+       keyIndexesNoTime,
+       fieldCompsRefs,
+       new Array[TypeSerializer[AnyRef]](0), //used only for object comparisons
+       booleanDirectionsNoTime)
+      
+    val collectionRowComparator = new CollectionRowComparator(rowComp)
+    
+    val inputCRowType = CRowTypeInfo(inputTypeInfo)
+ 
+    new RowTimeSortProcessFunction(
+      inputCRowType,
+      collectionRowComparator)
+
+  }
+  
+  
+  /**
+   * Function creates [org.apache.flink.streaming.api.functions.ProcessFunction] for sorting 
+   * elements based on proctime and potentially other fields
+   * @param collationSort The Sort collation list
+   * @param inputType input row type
+   * @param execCfg table environment execution configuration
+   * @return org.apache.flink.streaming.api.functions.ProcessFunction
+   */
+  private[flink] def createProcTimeSortFunction(
+    collationSort: RelCollation,
+    inputType: RelDataType,
+    inputTypeInfo: TypeInformation[Row],
+    execCfg: ExecutionConfig): ProcessFunction[CRow, CRow] = {
+
+    val keySortFields = getSortFieldIndexList(collationSort)
```

Comment:
I am not sure I understand what you mean. But I think the current setup to create the processfunction based on rowtime/ proctime are needed and is useful also for adding later the cases when we have offset/fetch. I would not like to modify this as it does not bring value (but it might be that I misunderstood what you mean)